src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -108,8 +108,9 @@ public int GetEditDistance(string target, int threshold = int.MaxValue)
         private static readonly ThreadLocal<int[,]> t_matrixPool = 
             new ThreadLocal<int[,]>(() => InitializeMatrix(new int[MaxMatrixPoolDimension, MaxMatrixPoolDimension]));
 
-        private static ThreadLocal<Dictionary<char, int>> t_dictionaryPool = 
-            new ThreadLocal<Dictionary<char, int>>(() => new Dictionary<char, int>());
+        private const int LastSeenIndexLength = 128;
+        private static ThreadLocal<int[]> t_lastSeenIndexPool = 
```

Comment:
:bulb: It may be possible to further improve this with a stack-allocated structure like this:

```csharp
private unsafe struct LastSeenIndex
{
    private const int LastSeenIndexLength = 128;
    private fixed byte data[LastSeenIndexLength];

    public byte this[int index]
    {
        get
        {
            if (index >= 0 && index < LastSeenIndexLength)
            {
                fixed (byte* data = this.data)
                {
                    return data[index];
                }
            }

            return 0;
        }

        set
        {
            if (index >= 0 && index < LastSeenIndexLength)
            {
                fixed (byte* data = this.data)
                {
                    data[index] = value;
                }
            }
        }
    }
}
```